pkg/storage/etcd.go

```diff
@@ -36,37 +37,77 @@ func NewEtcdStorage(endpoints []string) (*EtcdStorage, error) {
 }
 
 // Store stores the key
-func (etcds *EtcdStorage) Store(key string, value Data) error {
+func (etcds *EtcdStorage) StoreCheckin(key string, value time.Time) error {
 	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
-	encoded, _ := json.Marshal(value)
-	_, err := etcds.kv.Put(ctx, fmt.Sprintf("env=%s", key), string(encoded))
+	encoded, err := json.Marshal(value)
+	if err != nil {
+		cancel()
```

Comment:
Let's use `defer cancel()` right after the context is created.

A question that I have, is it needed the context to be initialized before marshaling or it could just be right before calling `Put`, which is where the context is used?